T30171 Rebase on 3.38.0

pwithnall opened the pull request:
Rebase the gnome-software code on the upstream 3.37.92 release (specifically, directly onto the `3.37.92` tag).

Not too much to point out in this rebase.
 * I haven’t dropped the Speedwagon/Desktop shortcut/Discovery feed support commits yet, as it was too much to do in a single rebase pass. I have highlighted the relevant commits in the report below, though, so they can easily be dropped in the next rebase pass (likely onto 3.38.0), or can be `git revert`ed separately.
 * I squashed in a lot of small fixup patches to the commits they are fixing.
 * The only significant code changes I remember having to make were CSS parsing performance improvements in “Add the new GsBackgroundTile” and “Add GsSideFilterRow widget”, so those two commits will need to be re-reviewed in detail.
 * Everything else is noted as “(Rebase 3.38: blah blah)” in the commit messages.
 * We’ve gone from 238 to 133 patches.
 * Packaging changes in https://github.com/endlessm/gnome-software/pull/556.

---

Full commit-by-commit notes:
 * *pick* `62f342587 Revert "Remove the now-unused code that supported the colorful category buttons"` — keep, EOS-specific
 * *pick* `a5e7450c4 core: Add USB category` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `72f35f15e flatpak: Select USB category when loading a folder that has a removable remote` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `faa4b9414 flatpak: Allow offline installation of apps that have the USB category` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `ba63bb2de Set gnome-software as default handler for x-content/ostree-repository` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `6cc764376 flatpak: Emit updates-changed when a USB remote is loaded` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `18558a09d lib: Make gs_app_{get,set}_priority() public` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `50a82b988 flatpak: Add collection ID support to GsFlatpakApp` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `09696a6e9 flatpak: Add apps to the USB category` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `125ad0313 flatpak: Reload plugin when a removable remote is unmounted` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `a2bce1f75 gs-plugin-job: Add copy-dest property` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `b0cbd319b gs-plugin-loader: Add vfuncs for copying an app` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `d03d6f730 gs-plugin-loader: Add monitoring of mounted USB drives` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `12184226d gs-plugin: Add new GS_PLUGIN_STATUS_COPYING status` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `852df6535 flatpak: Set flatpak::ExtraData for extra-data apps` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `930289d2f flatpak: implement copying an app to removable media` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `3a4a0a07f gs-page: Add gs_page_copy_app() to support copying apps to USB drives` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `e8c31bfa3 details page: restructure code to handle progress indicators` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `828c1e060 gs-details-page: Add a ‘Copy to USB’ button for copying apps` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `67ded7e5c Display the USB category even if the inserted disk has no apps` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `d12d8f964 Show empty state message for USB category for appless disks` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `a919205ca lib: Ignore USB sticks if they are read-only` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `a388582ec gs-plugin-loader: Add vfuncs for copying the OS` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `da2260d94 gs-page: Add gs_page_copy_os() to support copying the OS to USB drives` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `08026e352 Add 'copy OS to USB' button to USB category` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `28a863a08 src: Expose origin hostname in app details for blacklisting` — keep, EOS-specific
 * *pick* `847b61573 eos: Split blacklisting out of old eos plugin into new eos-blacklist plugin` — squashed in subsequent commits and significantly updated commit message to “eos: Add eos-blocklist plugin”
 * **fixup** `a7a82a5da fixup! eos: Split blacklisting out of old eos plugin into new eos-blacklist plugin` — squashed into “eos: Split blacklisting out …”
 * **fixup** `e2f931b27 eos-blacklist: Define gs_plugin_refine_app() not gs_plugin_refine()` — squashed into “eos: Split blacklisting out …”
 * **fixup** `84021edc9 fixup! eos-blacklist: Define gs_plugin_refine_app() not gs_plugin_refine()` — squashed into “eos: Split blacklisting out …”
 * **fixup** `c41161e22 eos-blacklist: Don’t blacklist gnome-system-monitor-kde.desktop` — squashed into “eos: Split blacklisting out …”
 * **fixup** `f0b51b7cb eos-blacklist: Drop blacklisting of renamed apps` — squashed into “eos: Split blacklisting out …”
 * **fixup** `27ba42b88 eos-blacklist: Drop OS compatibility blacklist` — squashed into “eos: Split blacklisting out …”
 * **fixup** `227d79f30 eos-blacklist: Drop legacy app blacklist` — squashed into “eos: Split blacklisting out …”
 * *pick* `cbd4b46f0 gs-prefs-dialog: Hide automatic updates setting when using Mogwai` — not yet upstreamed
 * *pick* `f75d93999 Add the top bar about auto updates to the Updates page` — not yet upstreamed
 * **fixup** `2483cc7c4 fixup! Add the top bar about auto updates to the Updates page` — squashed into “Add the top bar about auto updates to the Updates page”
 * *pick* `20443e4a2 eos: Hide the recent box` — keep, EOS-specific
 * *pick* `9260b062b Add EOS desktop plugin` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `57149ad83 eos: Disallow uninstallation of core applications` — keep, EOS-specific
 * *pick* `6122becb3 Add GsSideFilterRow widget` — keep, EOS-specific
 * *pick* `640965598 Add a side filter list to the shell` — keep, EOS-specific
 * *pick* `2761d7004 Set the side filter categories from the overview` — keep, EOS-specific
 * *pick* `9696c9b6e Keep the top tabs visible when in the category page` — keep, EOS-specific
 * *pick* `50555870e eos: Set the image tile CSS from a special metadata key` — keep, EOS-specific
 * *pick* `691ad6513 Use a GtkFlowBox for displaying the popular apps in the overview` — keep, EOS-specific
 * *pick* `721daac07 Adapt the app's grid in the overall/category views to the UI specs` — keep, EOS-specific
 * *pick* `74b9a4717 app: Add detailed signal for when metadata is updated on an app` — keep, EOS-specific
 * *pick* `af60aef07 eos: Download backgrounds for app thumbnails and cache them.` — keep, EOS-specific
 * **drop** `1b6b34d48 Prefer apps that have a thumbnail image to be shown as popular` — subsequently reverted
 * *pick* `9de96818e Remove DBusActivatable` — Speedwagon support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30293
 * *pick* `6cf014ab6 eos: Adapt existing categories to fit Endless users' needs` — keep, EOS-specific
 * *pick* `d4b9a90c0 updates: launch app page when clicking on update row` — discussion stalled upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/issues/743
 * **drop** `ccb5b648b Hide the license details in the details page` — subsequently reverted
 * *pick* `cb8a4a7c7 Change user-facing name from Software to App Center` — keep, EOS-specific
 * *pick* `c1b703f05 ui: Rename ‘Remove’ to ‘Uninstall’` — keep, EOS-specific
 * *pick* `e6556dbcd gs-details-page: Rename shortcut buttons to ‘Add to/Remove from Desktop’` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `de655580b Emphasize the "Add to Desktop" button` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `1a39fa28e eos: Support desktop icons for apps that use helper desktop files` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `1da3d3f7b Show the search button in the category and installed views` — keep, EOS-specific
 * **drop** `f5568cae7 eos: Use a proxy app for updating some apps together` — subsequently reverted
 * *pick* `e0afb2489 Disable the featured banner` — keep, EOS-specific
 * *pick* `bc4a6b88a Hide the "Editor's Picks" label in the featured page` — keep, EOS-specific
 * *pick* `0d6cf03ed eos: Fix launching Google Chrome` — keep, EOS-specific
 * *pick* `60420400e Sort popular apps alphabetically based on GSettings` — not yet upstreamed
 * *pick* `4ec080de4 feed: Add an EOS Discovery Feed Content Provider` — Discovery feed support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30294
 * *pick* `12e97b80a eos: Override certain icons when adding to desktop` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `1b53387b4 Add the new GsBackgroundTile` — keep, EOS-specific
 * *pick* `cc05993e0 Use the GsBackgroundTile in the overview and category pages` — keep, EOS-specific
 * *pick* `f68865517 updates: Set the refresh button insensitive when apps are updating` — not yet upstreamed
 * *pick* `0932c507b Use the speedwagon when opening GNOME Software` — Speedwagon support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30293
 * *pick* `c21fca4d4 Make the Add-to-desktop button the first button in the details page` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `0002acc90 Use GenericName in the desktop file to provide an alternative name` — keep, EOS-specific
 * *pick* `dbfdcc0ce Limit the number of certain ops in parallel to 1` — keep, EOS-specific
 * *pick* `1efb2bfd8 build: Override meson's default options` — keep, EOS-specific
 * *pick* `3c5b2aac9 Never show the first run dialog` — keep, EOS-specific
 * *pick* `18fc192b0 src: Re-style OS upgrade background and buttons` — keep, EOS-specific
 * *pick* `26de37947 Remove the big shopping bag icon from the loading page` — keep, EOS-specific
 * *pick* `1369695c8 flatpak: Set the apps' update priority` — keep, EOS-specific
 * *pick* `6658348a3 ui: Rename ‘Install’ to ‘Download’` — keep, EOS-specific
 * *pick* `fcf47782f gs-details-page: Label button as ‘Install’ not ‘Download’ for USB apps` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * **drop** `866fb3f59 eos: Include EknServicesMultiplexer in proxy app` — subsequently reverted
 * *pick* `eb05d58eb Do not invalidate OS upgrades in the Updates page` — not yet upstreamed
 * *pick* `10ede0e3c Display the USB icon in the background tile` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `527c081c5 eos: Make getting an app's desktop ID more robust` — keep, EOS-specific
 * *pick* `bba82ed41 flatpak: Don't disable the plugin if an installation setup succeeds` — pending review upstream as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/237
 * *pick* `1776b31cc flatpak: Refactor the plugin's setup method` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/237
 * *pick* `af57d0adc Add an "interactive" refine flag` — not yet upstreamed
 * *pick* `a17abdc76 Use the interactive flag in the details page's jobs` — not yet upstreamed
 * *pick* `1a9b57997 Lower the minimum number of popular apps required to be shown` — keep, EOS-specific
 * *pick* `9804fa9fd Save/load install queue using unique IDs` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `1047c86cd Fix showing the apps queued for installation in the installed page` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `3ecb32704 Try to install the apps that are queued after adding them` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `c54360557 Allow to cancel installation of apps that were queued for install` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `e6f7a424e Use a GsAppList for the pending apps` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `afaaf554d Fix order of saving apps that are queued install` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `daf2a2d2e Fix updating the installation queue on disk` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * **drop** `a842cb78f Do not re-create app tiles when switching categories` — subsequently reverted
 * **drop** `784f869b1 Remove apps from the categories cache when no longer present` — subsequently reverted
 * *pick* `a52eb6665 Always add apps that are to be installed to the install queue` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `88986a00f Fix how pending apps are installed from the installed page` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `8857040ec Display ongoing app-update at top of the list` — not yet upstreamed
 * *pick* `057666237 Allow to cancel pending installations that are queued` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `e05817244 Simplify how apps queued-for-install are cancelled` — Install queue support, pending upstreaming in https://phabricator.endlessm.com/T30642
 * *pick* `68413f1a0 flatpak: Return no-network error when failing to fetch remote metadata` — not yet upstreamed
 * *pick* `5f8b9eb90 eos: add support for copying OS` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `0d97445e4 eos: fix asking shell to add/remove "[Invalid UTF-8]"` — Desktop shortcut support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30446
 * *pick* `90fb7c620 eos: Check for OS updates on USB insertion` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `2a8fc1112 Only show "Copy to OS to USB" button if it might work` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * **drop** `eb5af9adf eos: split out processing of proxy app updates` — subsequently reverted
 * **drop** `c022fae1f eos: add proxy app for Endless Hack` — subsequently reverted
 * *pick* `47df2e5ef flatpak: Ignore FLATPAK_ERROR_ALREADY_INSTALLED when installing apps` — pending review upstream as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/247
 * *pick* `24562096e eos: Add EknServicesMultiplexer as a related app for EKN apps` — keep, EOS-specific
 * *pick* `334edb256 gs-appstream: Add support for Endless::HasDiscoveryFeedContent searches` — Discovery feed support; keeping for now and it can be reverted and dropped later as part of https://phabricator.endlessm.com/T30294
 * **drop** `a27e6555a eos: update list of proxied apps for the HackComponents` — subsequently reverted
 * **drop** `1e381c7d7 eos: update list of proxied apps for the HackComponents` — subsequently reverted
 * **drop** `c76902168 Add gs_app_clear_related method to the GsApp` — now unused
 * **drop** `a63c58020 Ask for related apps to be refined, in the Updates page` — proxy app support, no longer needed
 * **drop** `b716b0d09 flatpak: Refine proxy apps` — subsequently reverted
 * **drop** `a6959d67c eos: Add support for requiring proxy apps to install missing apps` — subsequently reverted
 * **drop** `b8d1c69e4 eos: Make the Hack proxy app install any missing apps` — subsequently reverted
 * *pick* `82d91e3eb add Learn to Code category` — keep, EOS-specific
 * *pick* `59d838e3e flatpak: Fix the ref kind when getting the main app of a related one` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/503
 * *pick* `44e285793 Select category correctly when "Back" is clicked` — keep, EOS-specific
 * **drop** `d27559787 overview: don't prefer popular apps with backgrounds` — effectively a revert of “Prefer apps that have a thumbnail image to be shown as popular”
 * **fixup** `21cdc4fdc eos: remove Minecraft from duplicated apps list` — squashed into “eos: Split blacklisting out …”
 * **drop** `e085b92b5 eos: update list of proxied apps for the HackComponents` — subsequently reverted
 * **fixup** `9bf03e7a5 eos: add more violent games to the list of banned apps` — squashed into “eos: Split blacklisting out …”
 * **fixup** `b716d32ac eos: ban violent apps for Solutions' Spark Kids Laptop images` — squashed into “eos: Split blacklisting out …”
 * **fixup** `25828306d eos: ban adult apps for Solutions' Spark Kids Laptop images` — squashed into “eos: Split blacklisting out …”
 * **fixup** `5b30e2abd eos: remove Scratch from duplicated apps list` — squashed into “eos: Split blacklisting out …”
 * *pick* `d7ae4ae0e appdata: remove release notes` — the gettext bug has been fixed, but this should be kept until release note translations are available upstream (they aren’t yet)
 * **fixup** `f4954688b eos: rename org.gnome.chess to org.gnome.Chess` — squashed into “eos: Split blacklisting out …”
 * **fixup** `327654d63 eos: remove de.billardgl.BillardGL from the Flathub hidden list` — squashed into “eos: Split blacklisting out …”
 * **fixup** `5ab4c1c31 eos: Ban violent/adult apps from OS products` — squashed into “eos: Split blacklisting out …”
 * **fixup** `a8536e121 eos-blacklist: Update ARM whitelist for renamed apps` — squashed into “eos: Split blacklisting out …”
 * *pick* `361358c25 details-page: Place "Uninstall" button below app screenshots` — keep, EOS-specific
 * *pick* `ba876f446 Replace desktop icon with Endless asset` — keep, EOS-specific
 * *pick* `55ee16871 gs-app-tile: Refresh the app tile when the app’s action changes` — keep, EOS-specific
 * *pick* `db15293d5 gs-app-tile: Refresh the app tile when the app’s metadata changes` — keep, EOS-specific
 * **fixup** `b1e2316a3 Use correct category ID "usb" not "USB"` — squashed into “eos: Split blacklisting out …”
 * *pick* `350740d55 eos-updater: Call the correct gs_plugin_refresh() implementation` — keep, EOS-specific
 * **fixup** `d4b0435e6 gs-overview-page: Update hash table when adding categories` — squashed into “Set the side filter categories from the overview”
 * *pick* `eb22438e2 flatpak: Properly populate USB category` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * **fixup** `ac8da2af2 eos-blacklist: Blacklist org.gnome.Evolution` — squashed into “eos: Split blacklisting out …”
 * *pick* `d43467455 flatpak: Introduce success/failure heuristics for app install and autoupdates` — pending review upstream at https://github.com/flatpak/flatpak/pull/3013
 * *pick* `bf7c90ecd Reload USB category on USB insertion` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * *pick* `4219a2d81 flatpak: Only show apps in the USB category` — USB support, pending upstreaming in https://phabricator.endlessm.com/T26487
 * **fixup** `0a313322c gs-shell: Avoid a warning passing NULL to gs_category_equal()` — squashed into “Select category correctly when "Back" is clicked”
 * *pick* `d3b1a784e updates monitor: Don't skip automatic updates that require new permissions` — keep, reworked upstream version of it has not yet landed in https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/329
 * *pick* `10139d485 appstream: Override <id> with <launchable type="desktop-id"> while parsing appdata files` — not yet upstreamed
 * **fixup** `f8a6fd488 eos-blacklist: Allow multiple locales for g-s apps` — squashed into “eos: Split blacklisting out …”
 * **fixup** `0dee9199e eos-blacklist: remove Etoys entry from duplicates list` — squashed into “eos: Split blacklisting out …”
 * *pick* `0c0117f55 HACK: Disable timeout for plugin job operations` — keep, EOS-specific
 * **fixup** `4e6462083 eos-blacklist: Cache call to flatpak_installation_get_default_locales` — squashed into “eos: Split blacklisting out …”
 * **fixup** `65739120f eos-blacklist: remove Frostwire from blacklist` — squashed into “eos: Split blacklisting out …”
 * **fixup** `e82eaf708 eos-blacklist: remove Eclipse from blacklist` — squashed into “eos: Split blacklisting out …”
 * **fixup** `e8ffaf4d9 eos-blacklist: remove PlatformIO from blacklist` — squashed into “eos: Split blacklisting out …”
 * **fixup** `013d688c1 eos-blacklist: remove Superpowers from blacklist` — squashed into “eos: Split blacklisting out …”
 * **drop** `03d2b120e transaction: Sync main app's state if a related ref is installed` — reworked upstream as “flatpak: Set apps to installed upon missing runtime install”
 * **drop** `df2a2a4f6 fixup! transaction: Sync main app's state if a related ref is installed` — squashed into “transaction: Sync main app's state if a related ref is installed”, then dropped
 * **drop** `9298ef8f5 transaction: Sync installed apps' state missing runtime is re-installed` — reworked upstream as “flatpak: Set apps to installed upon missing runtime install”
 * *pick* `a88669ba0 gs-plugin-loader: Show spinner only on app being copied to USB` — af2aea78936d8d82da039dea5f500bda3423a6e9
 * **fixup** `7799d9b03 fixup! gs-plugin-loader: Show spinner only on app being copied to USB` — squashed into “gs-plugin-loader: Show spinner only on app being copied to USB”
 * **fixup** `faedc2bf6 fixup! gs-plugin-loader: Show spinner only on app being copied to USB` — squashed into “gs-plugin-loader: Show spinner only on app being copied to USB”
 * **drop** `55abd923f eos: Remove proxy app for Hack apps` — subsequently reverted
 * **drop** `738b54aa1 eos: Remove unused code to get product name` — product name is needed by later changes to the blocklist plugin, so this was effectively reverted
 * **drop** `294e07a61 eos: Remove CompanionAppService from list of proxied platform apps` — subsequently reverted
 * **drop** `63a9a5547 eos: Fix app ids for platform proxied apps` — subsequently reverted
 * **drop** `f5a7f7f9a flatpak: Support updating proxy apps` — subsequently reverted
 * **drop** `3f8795393 flatpak: Show progress when updating proxy apps` — subsequently reverted
 * **drop** `330a2b209 eos: Remove unused code to install missing proxied apps` — subsequently reverted
 * **drop** `ba9be3fba eos: Remove unused code to get image version` — image version is needed by later changes to the blocklist plugin, so this was effectively reverted
 * **fixup** `1ca4f9a38 eos: Remove unused define` — squashed into “eos: Override certain icons when adding to desktop”
 * *pick* `be3de7d53 Integrate Transifex translation infrastructure` — keep, EOS-specific
 * **fixup** `74dc89479 gs-details-page: Avoid a NULL object dereference` — squashed into “Show spinner only on app being copied to USB”
 * **drop** `5350c44c2 Revert "Hide the license details in the details page"` — revert of an earlier commit
 * *pick* `e1ad493a3 eos: Add parental controls metrics on update poll` — keep, EOS-specific
 * **drop** `b5b840940 Revert "Remove apps from the categories cache when no longer present"` — revert of an earlier commit
 * **drop** `34d2c1f17 Revert "Do not re-create app tiles when switching categories"` — revert of an earlier commit
 * **drop** `daea49d0a gs-category-page: Block signal emission while reloading` — upstreamed as 4e42df8318b98b3fd27c1a9f392b4dd150a011fe
 * **drop** `fdbf92f9d gs-plugin-loader: Avoid calls to refine_app() and refine_wildcard()` — upstreamed as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/441
 * **drop** `b400d9003 gs-plugin-loader: Only do an early refine if sorting results` — upstreamed as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/441
 * **drop** `9c763ac59 core: Only define refine(), not refine_app(), where possible` — upstreamed as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/442
 * *pick* `ceb1df9ff Add translation support for several new languages` — keep, EOS-specific
 * **drop** `5f470166b Regenerate .po files` — translations will be reapplied later
 * **drop** `d8ba6a0c5 Reapply old downstream translations` — translations will be reapplied later
 * **drop** `939de8f99 Updates from Transifex` — translations will be reapplied later
 * **fixup** `492132cef Actually commit .po files.` — squashed into “Add translation support for several new languages”
 * **fixup** `69f6bc6e4 eos-blacklist: unblacklist Firefox from Flathub` — squashed into “eos: Split blacklisting out …”
 * **drop** `76f258cfa Revert "eos: Remove unused code to install missing proxied apps"` — revert of an earlier commit
 * **drop** `ac8e9aab6 Revert "eos: Fix app ids for platform proxied apps"` — revert of an earlier commit
 * **drop** `4d28adfdd Revert "eos: Remove CompanionAppService from list of proxied platform apps"` — revert of an earlier commit
 * **drop** `1d5101968 Revert "eos: Remove proxy app for Hack apps"` — revert of an earlier commit
 * **drop** `4021ae938 Revert "eos: update list of proxied apps for the HackComponents"` — revert of an earlier commit
 * **drop** `c9755e378 Revert "eos: Make the Hack proxy app install any missing apps"` — revert of an earlier commit
 * **drop** `b4e709186 Revert "eos: Add support for requiring proxy apps to install missing apps"` — revert of an earlier commit
 * **drop** `ba052601a Revert "eos: update list of proxied apps for the HackComponents"` — revert of an earlier commit
 * **drop** `bf6c2a9ba Revert "eos: update list of proxied apps for the HackComponents"` — revert of an earlier commit
 * **drop** `28873a5fa Revert "eos: add proxy app for Endless Hack"` — revert of an earlier commit
 * **drop** `20f43b94f Revert "eos: split out processing of proxy app updates"` — revert of an earlier commit
 * **drop** `affba30c0 Revert "eos: Include EknServicesMultiplexer in proxy app"` — revert of an earlier commit
 * **drop** `0ff6240d4 Revert "eos: Use a proxy app for updating some apps together"` — revert of an earlier commit
 * **drop** `012f398ac Revert "flatpak: Show progress when updating proxy apps"` — revert of an earlier commit
 * **drop** `b2b88ec6d Revert "flatpak: Support updating proxy apps"` — revert of an earlier commit
 * **drop** `aa799033f Revert "flatpak: Refine proxy apps"` — revert of an earlier commit
 * **fixup** `25cb9931b eos-blacklist: Don’t blacklist ‘evergreen’ apps` — squashed into “eos: Split blacklisting out …”
 * **drop** `a6d2f8e55 lib: Formalise use of GS_APP_PROGRESS_UNKNOWN` — upstreamed as a5d2bbbb8847f189f7bb594d8c1917adb1db85aa
 * **drop** `363453bbe flatpak: Endless-only change to depend on backported flatpak APIs` — dropped as flatpak 1.8 is now in Endless OS
 * *pick* `024c480da gs-debug: Ignore glib-networking debug messages` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/463
 * *pick* `7a73d5dbf flatpak: Block on download scheduler for updates` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/463
 * *pick* `899ed27d8 gs-metered: Fix ‘Since’ tags on documentation comments` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/463
 * *pick* `051094178 gs-metered: Remove Mogwai schedule entries when downloads are complete` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/463
 * *pick* `98bc9f159 shell: Ignore failure to hold the scheduler daemon` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/463
 * **fixup** `b38eb90c4 eos-blacklist: Rename to eos-blocklist` — squashed into “eos: Split blacklisting out …”
 * **drop** `c0de15d68 flatpak: Fix bundle installation` — upstreamed as 27c0b3f843cec4320de6b06ea6024fba0b210530
 * **drop** `fea4b6c3b flatpak: Show apps installed from bundles` — upstreamed as 41d1a6de4e85db8efe6deb0847b138534ae5b0e2
 * **drop** `c7d2f7a30 flatpak: Fix searching for installed bundles` — upstreamed as d306a601b2e0e39e8cc5391bdf22614020facf20
 * **drop** `dd911996a flatpak: Add a test for installing a bundle` — upstreamed as c42bad07e986d9aded4b3899dbaa3ea17cc7fb8b
 * **drop** `8c5856155 flatpak: Fix icons for installed bundles` — upstreamed as f58c6dbbadbd9ba0815bf2a548359c5f4eb2cd2e
 * **drop** `4b7ac5d85 tests: Avoid using a hard-coded directory for temporary test data` — upstreamed as 9f1f5ba2c3e9fff25bac20512523f29a502a6c7c
 * **drop** `3a1b68b39 flatpak: Use g_assert_true() in self tests` — upstreamed as b52aa1a88d00c785d0ecc69d50c2864dc34fc5fe
 * **drop** `3559d67d1 flatpak: Rename an internal function argument` — upstreamed as 99284d57f9c88757c1f9863e9db099d80e180492
 * **drop** `ebd46bc28 flatpak: Set app to installed when a related thing is updated` — upstreamed as dd100c88d0fe1dbdc271fb685cb34b5fe791c91c
 * **drop** `fbec8bbdf flatpak: Set apps to installed upon missing runtime install` — upstreamed as 87b8de8652588d8dba4e63e2f3f509f3a71c0695
 * **drop** `a081728c5 flatpak: Endless-only change to lower flatpak version checks` — dropped as flatpak 1.8 is now in Endless OS
 * **fixup** `230ecdc0f build: Enable unit tests for CI` — squashed into “build: Override meson's default options”
 * **fixup** `53e4a5d79 dummy: Skip hang search test` — squashed into “HACK: Disable timeout for plugin job operations”
 * *pick* `41eef1836 fedora-langpacks: Skip self tests under non-fedora distros` — pending review upstream at https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/495
 * *pick* `60261ae1a lib: Skip mogwai checks in unit tests` — not yet upstreamed
 * **fixup** `92a5d90e4 dummy: Use Endless category name` — squashed into “eos: Adapt existing categories to fit Endless users' needs”
 * **fixup** `58be7ed1d dummy: Fix self test failure` — squashed into “Always add apps that are to be installed to the install queue”
 * *pick* `11d1ea193 gs-app: Allow change from AVAILABLE_LOCAL to QUEUED_FOR_INSTALL` — not yet upstreamed
 * *pick* `71c7ff824 tests: Skip flatpak parental controls checks` — not yet upstreamed
 * **fixup** `0a709b425 lib: Fix state check for install-queue removal` — squashed into “Simplify how apps queued-for-install are cancelled”
 * *pick* `910b4ae2c gs-plugin-loader: Don't erroneously set apps to AVAILABLE` — not yet upstreamed
 * **drop** `cae2dcd42 tests: Fix racy assertion in flatpak test` — upstreamed as 4370c6a39788d3ac64969e7bef07923f59189aa4
 * *pick* `af998157f flatpak: Skip self tests on arm` — workaround, needs fixing upstream as https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1047

https://phabricator.endlessm.com/T30171

mwleeds commented:
On my machine if I try to build this in a toolbox container with `DEB_BUILD_OPTIONS=nocheck dpkg-buildpackage -uc -us -b` here's the tail end of the output:
```
make[1]: Entering directory '/sysroot/home/mwleeds/git/gnome-software'
dh_missing --fail-missing
dh_missing: warning: usr/lib/x86_64-linux-gnu/libgnomesoftware.a exists in debian/tmp but is not installed to anywhere 
	The following debhelper tools have reported what they installed (with files per package)
	 * dh_install: gnome-software (36), gnome-software-common (5), gnome-software-dev (2), gnome-software-doc (1), gnome-software-plugin-flatpak (2)
	 * dh_installdocs: gnome-software (2), gnome-software-common (0), gnome-software-dev (0), gnome-software-doc (0), gnome-software-plugin-flatpak (0)
	 * dh_installman: gnome-software (0), gnome-software-common (0), gnome-software-dev (0), gnome-software-doc (0), gnome-software-plugin-flatpak (0)
	If the missing files are installed by another tool, please file a bug against it.
	When filing the report, if the tool is not part of debhelper itself, please reference the
	"Logging helpers and dh_missing" section from the "PROGRAMMING" guide for debhelper (10.6.3+).
	  (in the debhelper package: /usr/share/doc/debhelper/PROGRAMMING.gz)
	Be sure to test with dpkg-buildpackage -A/-B as the results may vary when only a subset is built
	If the omission is intentional or no other helper can take care of this consider adding the
	paths to debian/not-installed.

	Remember to be careful with paths containing "x86_64-linux-gnu", where you might need to
	use a wildcard or (assuming compat 13+) e.g. ${DEB_HOST_MULTIARCH} in debian/not-installed
	to ensure it works on all architectures (see #961104).
dh_missing: error: missing files, aborting
make[1]: *** [debian/rules:97: override_dh_missing] Error 255
make[1]: Leaving directory '/sysroot/home/mwleeds/git/gnome-software'
make: *** [debian/rules:78: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
```

mwleeds commented:
> We’ve gone from 238 to 133 patches.

This merits some celebration, good job! 🎊

mwleeds commented:
> fixup b1e2316a3 Use correct category ID "usb" not "USB" — squashed into “eos: Split blacklisting out …”

Looks like this wasn't squashed, it's still there. And it should be squashed into at least these, possibly others as well:

- flatpak: Allow offline installation of apps that have the USB category
- flatpak: Add apps to the USB category

mwleeds commented:
I pulled the patch from #557 onto this branch.

mwleeds commented:
>pick 1776b31cc flatpak: Refactor the plugin's setup method — not yet upstreamed

No it's pending upstream in the same MR as the patch above it, already linked in the commit description

mwleeds commented:
>pick a63c58020 Ask for related apps to be refined, in the Updates page — keep, EOS-specific

The commit message says it's needed for proxy apps, so maybe should be dropped?

mwleeds commented:
>pick 03d2b120e transaction: Sync main app's state if a related ref is installed — not yet upstreamed

This should be dropped because it is obsoleted by the patch "flatpak: Set apps to installed upon missing runtime install". Sadly it also causes that newer patch not to work, because the `set_skipped_related_apps_to_installed()` call is only in the `INSTALL_BUNDLE` case rather than `INSTALL` also.

mwleeds commented:
>pick ba9be3fba eos: Remove unused code to get image version — image version is needed by later changes to the blocklist plugin, so this was effectively reverted

So drop then?

mwleeds commented:
>pick af998157f flatpak: Skip self tests on arm — pending review upstream as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/500

That MR is unrelated to that patch. We could upstream it, but it's more of a workaround than a fix, and there's [an issue open](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1047) about the real fix.

mwleeds commented:
Made it to the end of the branch. Looks good otherwise, thanks for your work on this.

pwithnall commented:
Rebased on the upstream `3.38.0` tag (it just adds some translation updates on top of 3.37.92, nothing else).

> > fixup [b1e2316](https://github.com/endlessm/gnome-software/commit/b1e2316a35286536609d3402c06778df10c0b6fa) Use correct category ID "usb" not "USB" — squashed into “eos: Split blacklisting out …”
> 
> Looks like this wasn't squashed, it's still there. And it should be squashed into at least these, possibly others as well:
> 
>     * flatpak: Allow offline installation of apps that have the USB category
>     * flatpak: Add apps to the USB category
>     * gs-details-page: Label button as ‘Install’ not ‘Download’ for USB apps
>     * Display the USB icon in the background tile

Oops, yes, split up and squashed.

> > pick [1776b31](https://github.com/endlessm/gnome-software/commit/1776b31cc5ed133368401a6e8e9a3373b90aa36e) flatpak: Refactor the plugin's setup method — not yet upstreamed
> 
> No it's pending upstream in the same MR as the patch above it, already linked in the commit description

Updated in the notes above.

> > pick [a63c580](https://github.com/endlessm/gnome-software/commit/a63c580203f3da06e55e3d57fedb714ba716e325) Ask for related apps to be refined, in the Updates page — keep, EOS-specific
> 
> The commit message says it's needed for proxy apps, so maybe should be dropped?

Good spot, dropped (and updated the notes above).

> > pick [03d2b12](https://github.com/endlessm/gnome-software/commit/03d2b120e5bce7f2431a1d5753eb75ac920f1037) transaction: Sync main app's state if a related ref is installed — not yet upstreamed
> 
> This should be dropped because it is obsoleted by the patch "flatpak: Set apps to installed upon missing runtime install". Sadly it also causes that newer patch not to work, because the `set_skipped_related_apps_to_installed()` call is only in the `INSTALL_BUNDLE` case rather than `INSTALL` also.
> 
> > pick [9298ef8](https://github.com/endlessm/gnome-software/commit/9298ef8f5a5b882ce88cc8a25cdc21fa7ba801e5) transaction: Sync installed apps' state missing runtime is re-installed — not yet upstreamed
> 
> As above, this should be dropped as it was made obsolete by "flatpak: Set apps to installed upon missing runtime install".

Both dropped and updated the notes above. Thanks.

> > pick [ba9be3f](https://github.com/endlessm/gnome-software/commit/ba9be3fba98a5d8563ad6fa7f4e2ac97b2f33118) eos: Remove unused code to get image version — image version is needed by later changes to the blocklist plugin, so this was effectively reverted
> 
> So drop then?

I had actually dropped it but forgot to update the notes. Updated them now.

> > pick [af99815](https://github.com/endlessm/gnome-software/commit/af998157fa403523d635224f0d7b0dc5673cb6ae) flatpak: Skip self tests on arm — pending review upstream as https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/500
> 
> That MR is unrelated to that patch. We could upstream it, but it's more of a workaround than a fix, and there's [an issue open](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1047) about the real fix.

Updated the notes (the commit’s log message was already correct).

pwithnall commented:
Aaand it looks like force-pushing an update has caused github to helpfully delete the older comments from Phaedrus :angry:

mwleeds commented:
> Aaand it looks like force-pushing an update has caused github to helpfully delete the older comments from Phaedrus angry

That is frustrating, but I take it you think they were addressed? I don't remember what they were.

mwleeds commented:
I don't think it makes sense to do any more in-depth review but the branch looks good at a high level. Feel free to merge.

pwithnall commented:
> That is frustrating, but I take it you think they were addressed? I don't remember what they were.

Yes, I addressed all the comments, and all (except one?) are represented in the quotes in [my reply](https://github.com/endlessm/gnome-software/pull/555#issuecomment-692624839), so it’s not the end of the world.
